.travis.yml

```diff
@@ -37,7 +37,8 @@ matrix:
 before_install:
   - "export DISPLAY=:99.0"
   - "sh -e /etc/init.d/xvfb start"
-  - npm config set spin false
+  # if npm version is less than 3.0.0, upgrade to 3
+  - if [[ $(npm -v | cut -d '.' -f 1) -lt 3 ]]; then npm i -g npm@^3; fi
```

Comment:
My take on this would be to avoid any complexity that's not needed. Just always installing the latest version would of course slow down the build a bit so maybe that's a reason not to do it? I'm not sure what the additional time would be really…